### PR TITLE
[proj4] remove feature "database"

### PR DESCRIPTION
**Pull request by autoantwort**
Update port and fix core build (there was no feature that you can turn off, it was always required (was default feature so nobody checked it))

### CODE REVIEW

File: ports/proj4/usage [hunk 1]
```diff
@@ -1,4 +1,4 @@
 The package proj4 provides CMake targets:
 
-    find_package(PROJ CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE PROJ::proj)
+    find_package(proj4 CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE PROJ4::proj)
```
Comment (dg0yt) [inline ports/proj4/usage:4]: Do you intentionally propose the legacy usage for the legacy name of PROJ?
This is the reference:
https://proj.org/development/cmake.html#
It should also apply to version 7.

**Review by dg0yt — COMMENTED** [review]
There is existing work at #16494, blocked by vtk's dependency on the legacy API which is removed in PROJ 8.

IMO there should be two features:
- tools (vcpkg policy)
- network (to opt-out from curl and crypto dependencies).

File: ports/proj4/usage (hunk 1 above)
Comment (dg0yt) [reply ports/proj4/usage:4]: PS: I would still recommend the target instead of the variables.

**Comment by JackBoosY** [comment]
```
Package proj4 does not have a database feature
Note: Updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.
```

File: ports/proj4/usage (hunk 1 above)
Comment (autoantwort) [reply ports/proj4/usage:4]: No this then was an error by me

**Comment by dg0yt** [comment]
> > ```
> > Package proj4 does not have a database feature
> > ```
> 
> I think this is a bug in the ci, or can you otherwise explain this error?

Port mapnik explicitly depends on proj[database]. Key proj data is stored in a sqlite database, so practically everyone needs it.

**Comment by autoantwort** [comment]
> > > ```
> > > Package proj4 does not have a database feature
> > > ```
> > 
> > 
> > I think this is a bug in the ci, or can you otherwise explain this error?
> 
> Port mapnik explicitly depends on proj[database]. Key proj data is stored in a sqlite database, so practically everyone needs it.

Aah. Yeah the project also does not build without database feature (there is no database feature you can turn off, it then simply fails to find the sqlite executable)

**Comment by JackBoosY** [comment]
@autoantwort 
https://github.com/microsoft/vcpkg/blob/c26101d71e1e3f8a8a010bf6854c983ab777b9e9/ports/mapnik/vcpkg.json#L152-L162

**Comment by JackBoosY** [comment]
When building python3:x64-osx:
```
install: mkdir /Users/vagrant/Data/packages/python3_x64-osx/Users/vagrant/Data/installed/x64-osx/debug/lib: File exists
make: *** [altbininstall] Error 71
make: *** Waiting for unfinished jobs....
```
I saw this error before, is that related to your changes?

**Comment by dg0yt** [comment]
> When building python3:x64-osx:
> 
> ```
> install: mkdir /Users/vagrant/Data/packages/python3_x64-osx/Users/vagrant/Data/installed/x64-osx/debug/lib: File exists
> make: *** [altbininstall] Error 71
> make: *** Waiting for unfinished jobs....
> ```
> 
> I saw this error before, is that related to your changes?

@JackBoosY That's the error we saw in https://github.com/microsoft/vcpkg/pull/19540#issuecomment-927554949

**Comment by autoantwort** [comment]
> I saw this error before, is that related to your changes?

 No. I also get this error local all the time when I want to build python3.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Already tested all features successfully on `x86-windows` and `x64-windows-static`.

**Comment by BillyONeal** [comment]
Thanks for the bugfix!

**Pull request merged**